consul/resource_service_defaults_config_entry.go

```diff
@@ -0,0 +1,557 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package consul
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+	"strings"
+
+	consulapi "github.com/hashicorp/consul/api"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+var upstreamConfigSchema = &schema.Resource{
+	Schema: map[string]*schema.Schema{
+		"name": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"partition": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"namespace": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"peer": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"envoy_listener_json": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"envoy_cluster_json": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"protocol": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"connect_timeout_ms": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+		"limits": {
+			Type:     schema.TypeSet,
+			Optional: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"max_connections": {
+						Type:     schema.TypeInt,
+						Optional: true,
+					},
+					"max_pending_requests": {
+						Type:     schema.TypeInt,
+						Optional: true,
+					},
+					"max_concurrent_requests": {
+						Type:     schema.TypeInt,
+						Optional: true,
+					},
+				},
+			},
+		},
+		"passive_health_check": {
+			Type:     schema.TypeSet,
+			Optional: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"interval": {
+						Type:     schema.TypeInt,
+						Optional: true,
+					},
+					"max_failures": {
+						Type:     schema.TypeInt,
+						Optional: true,
+					},
+					"enforcing_consecutive_5xx": {
+						Type:     schema.TypeInt,
+						Optional: true,
+					},
+					"max_ejection_percent": {
+						Type:     schema.TypeInt,
+						Optional: true,
+					},
+					"base_ejection_time": {
+						Type:     schema.TypeInt,
+						Optional: true,
+					},
+				},
+			},
+		},
+		"mesh_gateway": {
+			Type:     schema.TypeSet,
+			Optional: true,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"mode": {
+						Type:     schema.TypeString,
+						Optional: true,
+					},
+				},
+			},
+		},
+		"balance_outbound_connections": {
+			Type:     schema.TypeString,
+			Optional: true,
+		},
+	},
+}
+
+var serviceDefaultsConfigEntrySchema = map[string]*schema.Schema{
+	"kind": {
```

Comment:
Is `kind` still required given that we are moving away from the `config_entry` resource? Wouldn't it always be `service-defaults`?

Reply:
It is computed. User can not enter this. Required for linke `474` to set kind.